api/controller/auth.js

```diff
@@ -176,8 +176,34 @@ exports.login = function(req, res) {
  **/
 exports.signup = function(req, res) {
 
-	//insert the user using the same logic but without auth
-	users.addUser(req, res);
+	var user = JSON.parse(req.body.user);
+	if(user.beforeSignup) {
+		validateUsername(user.name, function(unique) {
+			if(unique) {
+				res.send(true);
+			} else {
+				res.status(401).send({
+					'error': 'User with same name already exist',
+					'code': 22
+				});
+			}
+		});
+	} else {
+		//insert the user using the same logic but without auth
+		users.addUser(req, res);
+	}
+};
+
+function validateUsername(name, callback) {
+	users.getAllUsers({
+		'name': name
```

Comment:
Use string with regex similar to this instead of directly passing the name.

https://github.com/llaske/sugarizer-server/blob/a5232f33f28341173fcf551529555c342e4ad08b/api/controller/auth.js#L68-L70

Reply:
Added this in b7343ef.